connaisseur/keys.py

```diff
@@ -0,0 +1,141 @@
+import base64
+import hashlib
+import re
+import binascii
+
+import ecdsa
+import rsa
+from pyasn1.error import PyAsn1Error
+
+from connaisseur.image import Image
+from connaisseur.exceptions import NoSuchClassError, WrongKeyError
+
+KMS_REGEX = r"(?:awskms|gcpkms|azurekms|hashivault|k8s):\/{2,3}[a-zA-Z0-9_.+\/:-]+"
+KEYLESS_REGEX = r"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
+
+
+class Key:
+    """
+    Abstract Key class, containing the public key used for verifying the image signatures.
```

Comment:
The class does not necessarily contain the public key, does it? Afaik the idea of KMS is to have a reference, not the key itself, so this is probably false.

Does the key abstraction necessitate that the key is a public one? Couldn't, using KMS, there be an HMAC key?

Maybe rather in the direction of `Key object represents a key and can be used in conjunction with a validator to verify objects in that validator's domain.`

Reply:
Additionally from [PEP257](https://www.python.org/dev/peps/pep-0257/): `The docstring for a class should summarize its behavior and list the public methods and instance variables. If the class is intended to be subclassed, and has an additional interface for subclasses, this interface should be listed separately (in the docstring).`

Not sure what exactly that entails, but I would highlight that each subclass needs to implement verify, which kind of ties into making it an abstractmethod as per comment above

Reply:
Seeing Keyless, it doesn't even represent a Key, maybe the name is not correct in the first place. Maybe TrustAnchor?